File: matlab/ca_analyzeImage/analyzeImage.m
function analyzeImage( inputFile, outputFile )

  % read files from the list
  fid = fopen(inputFile,'r');
  C = textscan(fid,'%s');
  fclose(fid);

  % create output file
  ofid = fopen(outputFile,'w');
  
  filenames = C{1};
   
  % first file of the list
  I = imread(char(filenames(1)));
  [d1,h1] = basicInfo(I);  
  d2 = [];
  h2 = [];
  %[d2,h2] = GLCMInfo(I,2);
  data = [d1 d2];
  header = [h1 h2];

  % print header
  fprintf(ofid,'Filename');
  for j=1:length(header)
      fprintf(ofid,',%s',char(header(j)));
  end
  fprintf(ofid,'\n');
  
  % print data
  fprintf(ofid,'%s',char(filenames(1)));
  for j=1:length(data)
      fprintf(ofid,',%.4f',data(j));
  end
  fprintf(ofid,'\n');
  
  % for each input file
  for i=2:length(filenames)
      I = imread(char(filenames(i)));
      [d1] = basicInfo(I);
      %[d2] = GLCMInfo(I,2);
      d2 = [];
      data = [d1 d2];
      fprintf(ofid,'%s',char(filenames(i)));
      for j=1:length(data)
        fprintf(ofid,',%.4f',data(j));
      end
      fprintf(ofid,'\n');
  end
  
  % close the output file
  fclose(ofid);
  
end

File: R/analyzeImage.R

# load features
source("basicInfo.R")
source("EBImageFeatures.R")
source("colorFeatures.R")

# set default values
listFile <- "sample.txt"
resultFile <- "results.txt"
logFile <- "log.txt"

# load arguments
cargs <- Sys.getenv(c('listFile','resultFile','logFile'))

if (nchar(cargs[1]) > 0) listFile <- cargs[1];
if (nchar(cargs[2]) > 0) resultFile <- cargs[2];
if (nchar(cargs[3]) > 0) logFile <- cargs[3]

# read list of filenames
fin <- file(listFile,"r")
files <- readLines(fin)
n <- length(files)
close(fin)

# create output
fon <- file(resultFile,"w")


# for each file
for(i in 1:n) {
  f <- files[i]

  v1 <- basicInfo(f)
  v2 <- EBImageFeatures(f)
  v3 <- colorFeatures(f)

  v <- c(v1,v2,v3)

  # also print header if it's the first line
  if (i == 1) {
     cat(names(v),"\n",file=fon,sep=",")
     cat("string",rep("float",length(v)),"\n",file=fon,sep=",")
  }
       
  # print feature values
  cat(f,v,"\n",file=fon,sep=",")
}

# close output
close(fon)